### PR TITLE
Copter: factor out a mode-change-failed method

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

**Comment by peterbarker** [comment]
This was originally written before AP_Vehicle was a thing.

I wonder if we should put this method on AP_Vehicle rather than just on Copter?

**Review by khancyr — APPROVED** [review]
Yep, moving this to AP_Vehicle would make sens, but we don't have unified mode handling yet so, it could be done in another PR.

**Review by tridge — APPROVED** [review]

File: ArduCopter/mode.cpp [hunk 1]
```diff
@@ -177,6 +177,12 @@ Mode *Copter::mode_from_mode_num(const Mode::Number mode)
 }
 
 
+void Copter::mode_change_failed(const Mode *mode, const char *reason)
```
Comment (rmackay9) [inline ArduCopter/mode.cpp:180]: can you put a comment above the method?

**Review by rmackay9 — APPROVED** [review]
just add the comment above the method and then feel free to merge..

File: ArduCopter/mode.cpp (hunk 1 above)
Comment (peterbarker) [reply ArduCopter/mode.cpp:181]: Done

**Pull request merged**